massa-bootstrap/src/server_binder.rs

```diff
@@ -270,6 +290,102 @@ impl BootstrapServerBinder {
         .deserialize::<DeserializeError>(&msg_bytes)
         .map_err(|err| BootstrapError::GeneralError(format!("{}", err)))?;
 
-        Ok(msg)
+        dbg!("SERVER received message");
+        Ok(dbg!(msg))
+    }
+
+    fn write_helper(
+        &mut self,
+        buf: &[u8],
+        duration: Option<Duration>,
+    ) -> Result<(), BootstrapError> {
+        self.poll
+            .registry()
+            .reregister(&mut self.duplex, BINDING_EVENT, Interest::WRITABLE)?;
+        let mut written = 0;
+
+        loop {
+            self.poll
+                .poll(&mut self.events, duration)
+                .map_err(|e| match e.kind() {
+                    ErrorKind::TimedOut | ErrorKind::WouldBlock => BootstrapError::TimedOut(e),
+                    _ => BootstrapError::IoError(e),
+                })?;
+            for event in self.events.iter() {
+                if event.token() == BINDING_EVENT {
+                    match self.duplex.write(&buf[written..]) {
+                        Ok(0) => {
+                            return Err(BootstrapError::IoError(std::io::Error::new(
+                                ErrorKind::WriteZero,
+                                "failed to write whole buffer",
+                            )))
+                        }
+                        Ok(n) => {
+                            written += n;
+                            if written == buf.len() {
+                                return Ok(());
+                            }
+                        }
+                        Err(ref e) if e.kind() == ErrorKind::WouldBlock => {}
+                        Err(ref e) if e.kind() == ErrorKind::TimedOut => {}
+                        Err(ref e) if e.kind() == ErrorKind::Interrupted => {}
+                        Err(e) => return Err(BootstrapError::IoError(e)),
+                    }
+                } else {
+                    panic!("unexpected token");
+                }
+            }
+        }
+    }
+    fn read_helper(
+        &mut self,
+        buf: &mut [u8],
+        duration: Option<Duration>,
+    ) -> Result<(), BootstrapError> {
+        self.poll
+            .registry()
+            .reregister(&mut self.duplex, BINDING_EVENT, Interest::READABLE)?;
+        let mut read = 0;
+
+        loop {
+            self.poll
+                .poll(&mut self.events, duration)
+                .map_err(|e| match e.kind() {
+                    ErrorKind::TimedOut | ErrorKind::WouldBlock => BootstrapError::TimedOut(e),
+                    _ => BootstrapError::IoError(e),
+                })?;
+            if self.events.is_empty() {
+                return Err(BootstrapError::TimedOut(std::io::Error::new(
+                    ErrorKind::TimedOut,
+                    "timed out",
+                )));
+            }
+            for event in self.events.iter() {
+                if event.token() == BINDING_EVENT {
```

Comment:
here, when you listen for your stream, you can receive `BINDING_EVENT` for other TcpStream ?

Reply:
As it is, the code will only listen for an event from the tcp stream assosciated with this particular server binding.